hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/dog/RoutineEnvironmentTransformations.scala

```diff
@@ -0,0 +1,414 @@
+    val dogPark = rawRecord.getOptionalBoolean("de_dogpark_yn")
+    // ONLY if: de_dogpark_yn = 1
+    if (dogPark.contains(true)) {
+      val dogparkTravel = rawRecord.get("de_dogpark_get_to")
+      val dogparkTravelOther = dogparkTravel.map(_.contains("98"))
+      dog.copy(
+        deDogpark = dogPark,
+        deDogparkDaysPerMonth = rawRecord.getOptionalNumber("de_dogpark_freq"),
+        deDogparkTravelWalk = dogparkTravel.map(_.contains("1")),
+        deDogparkTravelDrive = dogparkTravel.map(_.contains("2")),
+        deDogparkTravelBike = dogparkTravel.map(_.contains("3")),
+        deDogparkTravelPublicTransportation = dogparkTravel.map(_.contains("4")),
+        deDogparkTravelOther = dogparkTravelOther,
+        //ONLY if: de_dogpark_get_to(98) = '1'
+        deDogparkTravelOtherDescription = if (dogparkTravelOther.contains(true)) {
+          rawRecord.getOptional("de_dogpark_get_to_other")
+        } else None,
+        // Compute using hour & min columns, Multiply hours by 60 to get minutes
+        deDogparkTravelTimeMinutes = for {
+          hours <- rawRecord.getOptionalNumber("de_dogpark_hr")
+          minutes <- rawRecord.getOptionalNumber("de_dogpark_min")
+        } yield {
+          hours * 60 + minutes
+        }
+        //deDogparkTravelTimeMinutes = calculateTotalTime(
+        //  rawRecord.getOptionalNumber("de_dogpark_hr"),
+        //  rawRecord.getOptionalNumber("de_dogpark_min")
+        //)
+      )
+    } else {
+      dog.copy(
+        deDogpark = dogPark
+      )
+    }
+  }
+
+  /* Recreational Spaces */
+  def mapRecreationalSpaces(
+    rawRecord: RawRecord,
+    dog: HlesDogRoutineEnvironment
+  ): HlesDogRoutineEnvironment = {
+    val dogRecSpaces = rawRecord.getOptionalBoolean("de_spaces_yn")
+    // ONLY if: de_spaces_yn = 1
+    if (dogRecSpaces.contains(true)) {
+      val dogRecSpacesTravel = rawRecord.get("de_spaces_get_to")
+      val dogRecSpacesTravelOther = dogRecSpacesTravel.map(_.contains("98"))
+      dog.copy(
+        deRecreationalSpaces = dogRecSpaces,
+        deRecreationalSpacesDaysPerMonth = rawRecord.getOptionalNumber("de_spaces_freq"),
+        deRecreationalSpacesTravelWalk = dogRecSpacesTravel.map(_.contains("1")),
+        deRecreationalSpacesTravelDrive = dogRecSpacesTravel.map(_.contains("2")),
+        deRecreationalSpacesTravelBike = dogRecSpacesTravel.map(_.contains("3")),
+        deRecreationalSpacesTravelPublicTransportation = dogRecSpacesTravel.map(_.contains("4")),
+        deRecreationalSpacesTravelOther = dogRecSpacesTravelOther,
+        //ONLY if: de_spaces_get_to(98) = '1'
+        deRecreationalSpacesTravelOtherDescription = if (dogRecSpacesTravelOther.contains(true)) {
+          rawRecord.getOptional("de_spaces_get_to_other")
+        } else None,
+        // Compute using hour & min columns, Multiply hours by 60 to get minutes
+        deRecreationalSpacesTravelTimeMinutes = for {
+          hours <- rawRecord.getOptionalNumber("de_spaces_hr")
```

Comment:
Same comment about hours and minutes here. They should probably be split apart in the schema since the hours value comes from a dropdown.